lib/route/seg6.c

```diff
@@ -0,0 +1,76 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+
+#include <stdbool.h>
+#include <linux/seg6.h>
+#include <linux/ipv6.h>
+#include <netlink/utils.h>
+#include <netlink-private/utils.h>
+
+bool seg6_validate_srh(struct ipv6_sr_hdr *srh, int len, bool reduced)
```

Comment:
this function is taken from kernel sources, where it's licensed as GPL-2.0-or-later.

Here the license should be LGPL-2.1-only, which is incompatible.

The code seems not trivial enough to just ignore the problem. It also is not clear how to re-implement it, without looking at the original code (clean-room implementation). Maybe it would be possible, to ask the original authors whether they would agree to allow using their code under LGPL-2.1-only, so that it can be used in libnl3. The original authors possibly can be found with `git log -L'/seg6_validate_srh/,/^}$/:net/ipv6/seg6.c'` in the kernel sources.

Reply:
it also needs a code comment, that indicates where the code was copied from.

Reply:
Code comments added.

Not only seg6_validate_srh() is trivial, but some other funcitions:
**Author: David Lebrun <david.lebrun@uclouvain.be>**
lib/route/nh_encap_seg6.c: seg6_encap_parse_msg,
lib/route/seg6.c: seg6_validate_srh, seg6_dump_srh
lib/route/nh_encap_seg6_local.c: parse_nla_srh, put_nla_srh, parse_nla_action, seg6_local_encap_parse_msg
**Author: Andrea Mayer <andrea.mayer@uniroma2.it>**
lib/route/nh_encap_seg6_local.c: parse_nla_counters, seg6_local_fill_nla_counters, put_nla_counters, seg6_chk_next_csid_cfg, seg6_parse_nla_next_csid_cfg, parse_nla_flavors, put_nla_flavors, parse_nla_optional_attrs
**Author: Mathieu Xhonneux <m.xhonneux@gmail.com>**
lib/route/nh_encap_seg6_local.c: parse_nla_bpf, put_nla_bpf

I think after resolving other issues with this PR we should ask those people to sign commit